### PR TITLE
.NET 6 Cross platform

### PR DESCRIPTION
**Pull request by Rans4ckeR**
- Updated frameworks
  - .NET 4.8
  - .NET 6.0 (supports Windows 7 and up only)
- Switch to new project file format
- Includes .editorconfig with my default warning rules. This triggers many new warnings. The rules can be reviewed and updated, or I can fix the current warnings in another PR.
- Produces local NuGet package Rampastring.Tools
- Introduces platform safe SafePath methods for handling file system path operations.
- Removed WinForms dependency, can now be used cross platform.

API change:
```C#
bool FlashWindowEx(Form form)
```
is now
```C#
bool FlashWindowEx(IntPtr windowHandle)
```

### CODE REVIEW

File: Rampastring.Tools.csproj [hunk 1]
```diff
@@ -1,74 +1,40 @@
...
+    <UseWindowsForms>true</UseWindowsForms>
+    <Title></Title>
+    <Description>Rampastring's Generally Useful Library</Description>
+    <Company>Rampastring</Company>
+    <Product></Product>
+    <Copyright>Copyright © Rami 'Rampastring' Pasanen 2016-2021</Copyright>
+    <Trademark></Trademark>
+    <Version>1.1.0.0</Version>
+    <FileVersion>1.1.0.0</FileVersion>
+    <InformationalVersion>1.1.0.0</InformationalVersion>
```
Comment (Rampastring) [inline Rampastring.Tools.csproj]: In `AssemblyInfo` I used wildcard versioning to let Visual Studio automatically generate the last two integers of the version number. Is the same doable here?

File: Rampastring.Tools.csproj [hunk 2]
```diff
@@ -1,74 +1,40 @@
...
-    <Compile Include="INIProperties\StringProperty.cs" />
-    <Compile Include="IniSection.cs" />
-    <Compile Include="Logger.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
-    <Compile Include="Utilities.cs" />
-    <Compile Include="WindowFlasher.cs" />
+    <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
```
Comment (Rampastring) [inline Rampastring.Tools.csproj:35]: I'd prefer it without StyleCop as I have my own established code style partially using the VS document formatter and mostly find such analyzers an annoyance.

File: Rampastring.Tools.csproj (hunk 1 above)
Comment (Rans4ckeR) [reply Rampastring.Tools.csproj]: Personally I leave these fields as is, ignoring the actual values. Instead I let the build/publish workflow generate a new version based on previous commits etc. I then also have packages automatically uploaded to NuGet and GitHub. See [GitHub Action GitTools](https://github.com/marketplace/actions/gittools) for versioning. [Example](https://github.com/Rans4ckeR/RS.Fritz.Manager/blob/master/.github/workflows/dotnet.yml)

File: Rampastring.Tools.csproj (hunk 2 above)
Comment (Rans4ckeR) [reply Rampastring.Tools.csproj:35]: Perhaps you can export your existing rules to an .editorconfig file? VS can generate it based on an existing solution.

**Review by Rampastring — APPROVED** [review]

**Pull request merged**